tests/checker/test_https.py

```diff
@@ -15,34 +15,50 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 """
-Test news checking.
+Test https.
 """
-import pytest
+import subprocess
 
 from tests import need_network
-from . import LinkCheckTest
+from .httpserver import HttpsServerTest, CookieRedirectHttpRequestHandler
+from .. import get_file
 
+from linkcheck import httputil
 
-class TestHttps (LinkCheckTest):
+from OpenSSL import crypto
+
+class TestHttps(HttpsServerTest):
     """
     Test https: link checking.
     """
 
-    @pytest.mark.xfail
+    def __init__(self, methodName='runTest'):
+        super(TestHttps, self).__init__(methodName=methodName)
+        self.handler = CookieRedirectHttpRequestHandler
+        temp_key_file = get_file("key.pem")
+        subprocess.check_output(["openssl", "req", "-x509", "-newkey",
```

Comment:
That should have been `subprocess.check_call` rather than `subprocess.check_output`.